webpack/scenes/ContentViews/Details/Versions/BulkDelete/Steps/ReviewEnvironments.js

```diff
@@ -0,0 +1,103 @@
+  TableVariant,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import WizardHeader from '../../../../components/WizardHeader';
+import ActionSummary from '../ActionSummary';
+import { BulkDeleteContext } from '../BulkDeleteContextWrapper';
+import {
+  getNumberOfActivationKeys,
+  getNumberOfEnviroments,
+  getNumberOfHosts,
+  getVersionListString,
+} from '../BulkDeleteHelpers';
+
+export default () => {
+  const columnHeaders = [
+    __('Environment'),
+    __('Hosts'),
+    __('Activation keys'),
+    __('Associated Version'),
+  ];
+
+  const { versions } = useContext(BulkDeleteContext);
+  const numberOfActivationKeys = getNumberOfActivationKeys(versions);
+  const numberOfHosts = getNumberOfHosts(versions);
+  const affectedVersions = versions.filter(({ environments }) => !!environments.length);
+  const pluralVersions =
+    affectedVersions.length > 1;
+  const versionsListString = getVersionListString(affectedVersions);
+  const pluralEnvironments = getNumberOfEnviroments(versions) > 1;
+  const willBeDeleteText = pluralEnvironments ?
+    __(' will be removed from the listed environments and will no longer be available for promotion.') :
+    __(' will be removed from the listed environment and will no longer be available for promotion.');
+
+  return (
+    <>
+      <WizardHeader
+        title={pluralEnvironments ?
+          __('Review affected environments') :
+          __('Review affected environment')}
+      />
+
+      <ExpandableSection
+        toggleTextCollapsed={__('Show affected environments')}
+        toggleTextExpanded={__('Hide affected environments')}
+      >
+        <TableComposable variant={TableVariant.compact}>
+          <Thead>
+            <Tr>
+              {columnHeaders.map(col =>
+                <Th key={col}>{col}</Th>)}
+            </Tr>
+          </Thead>
+          <Tbody>
+            {versions?.map(({ environments, version }) => environments.map(({
+              id, name, activation_key_count: akCount,
+              host_count: hostCount,
+            }) =>
+            (
+              <Tr key={`${name}_${id}`}>
+                <Td>
+                  {name}
+                </Td>
+                <Td>{hostCount}</Td>
+                <Td>{akCount}</Td>
+                <Td>{`${__('Version')} ${version} `}</Td>
+              </Tr>
+            )))
+            }
+          </Tbody>
+        </TableComposable>
+      </ExpandableSection>
+      <ActionSummary
+        text={(pluralVersions ? __('Versions ') : __('Version ')) + versionsListString + willBeDeleteText
+        }
+      />
+      {!!(numberOfActivationKeys + numberOfHosts) &&
+        <TextContent>
+          {__('Required steps:')}
+        </TextContent>}
+      {!!numberOfActivationKeys &&
+        <ActionSummary
+          text={`${numberOfActivationKeys} ${numberOfActivationKeys > 1 ? __(' activation keys will need to be reassigned.') : __(' activation key will need to be reassigned.')}`}
```

Comment:
:eyes: https://github.com/Katello/katello/blob/a12116ef4cb843c87ca9f149b1b573cd14afb03c/webpack/utils/helpers.js#L113

Reply:
How would we use this?  
Like this? 
`__(` ${pluralize(numberOfActivationKeys, 'activation key')}  will need to be reassigned.`)`

Reply:
yeah! It mirrors the one available in Rails :)

Reply:
For posterity, this __( `${pluralize(numberOfActivationKeys, 'activation key')} will need to be reassigned.`) won't work for the translation function. 

What should instead be used for adding to string literals within translations is the "[FormattedMessage](https://formatjs.io/docs/react-intl/components/#usage)" component from react-intl.

Reply:
I added changes to address this.. not pretty but functional.